<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1158</width>
    <height>734</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0,0,0">
    <property name="topMargin">
     <number>9</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="1" column="6" rowspan="2">
     <layout class="QGridLayout" name="motor_gridLayout" columnstretch="1,1,50,50">
      <item row="5" column="1" colspan="3">
       <widget class="QComboBox" name="statusreturnlevel_comboBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>Always</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Read</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Never</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="16" column="2">
       <widget class="QDial" name="position_dial">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="accessibleDescription">
         <string/>
        </property>
        <property name="notchesVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_10">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Firmware</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>ID</string>
        </property>
       </widget>
      </item>
      <item row="12" column="2">
       <widget class="QDial" name="minimum_dial">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="notchesVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="16" column="3">
       <widget class="QDial" name="goal_dial">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="notchesVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="1" colspan="3">
       <widget class="QComboBox" name="baud_comboBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <item>
         <property name="text">
          <string>1000000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>500000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>400000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>250000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>200000</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>117647</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>57142</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>19230</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>9615</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="9" column="2" colspan="2">
       <widget class="QSlider" name="torquemax_horizontalSlider">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
        <property name="tickInterval">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="8" column="0" colspan="4">
       <widget class="QLabel" name="label_13">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Motor Limits:</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0" rowspan="2" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>18</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Angle limits</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Baudrate</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_8">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>return delay time</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="label_12">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Torque max</string>
        </property>
       </widget>
      </item>
      <item row="20" column="0" colspan="4">
       <widget class="QCheckBox" name="torque_checkBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="accessibleName">
         <string>xcfx</string>
        </property>
        <property name="text">
         <string>Torque</string>
        </property>
        <property name="tristate">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="18" column="3">
       <widget class="QLabel" name="label_4">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Goal</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1" colspan="3">
       <widget class="QLabel" name="firmware_label">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_9">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Status return level</string>
        </property>
       </widget>
      </item>
      <item row="12" column="3">
       <widget class="QDial" name="maximum_dial">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="notchesVisible">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="QLabel" name="label_11">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Motor Configuration:</string>
        </property>
       </widget>
      </item>
      <item row="13" column="2">
       <widget class="QLCDNumber" name="minimum_lcd">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="13" column="3">
       <widget class="QLCDNumber" name="maximum_lcd">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="2" colspan="2">
       <widget class="QSlider" name="returndelaytime_horizontalSlider">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximum">
         <number>508</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
        <property name="tickInterval">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="3">
       <widget class="QSpinBox" name="id_spinBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="17" column="2">
       <widget class="QLCDNumber" name="position_lcd">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="15" column="0" colspan="4">
       <widget class="Line" name="line_2">
        <property name="lineWidth">
         <number>5</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="18" column="2">
       <widget class="QLabel" name="label_3">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Current</string>
        </property>
       </widget>
      </item>
      <item row="17" column="3">
       <widget class="QLCDNumber" name="goal_lcd">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QLCDNumber" name="torque_lcd">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="16" column="0" rowspan="3" colspan="2">
       <widget class="QLabel" name="position_label">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>18</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Position</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLCDNumber" name="returndelaytime_lcd">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="smallDecimalPoint">
         <bool>false</bool>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Dec</enum>
        </property>
       </widget>
      </item>
      <item row="10" column="1" colspan="2">
       <widget class="Line" name="line_3">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="14" column="0" colspan="3">
       <widget class="QPushButton" name="eeprom_button">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>24</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Update EEPROM</string>
        </property>
       </widget>
      </item>
      <item row="14" column="3">
       <widget class="QPushButton" name="reset_button">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>24</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="2" rowspan="2">
     <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0" columnstretch="5,0" columnminimumwidth="0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetNoConstraint</enum>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="11" column="1">
       <widget class="QSpinBox" name="minimum_spinBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>253</number>
        </property>
        <property name="value">
         <number>91</number>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QLabel" name="id_label">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>ID ranges to search:</string>
        </property>
       </widget>
      </item>
      <item row="18" column="0" colspan="2">
       <widget class="QProgressBar" name="scan_progressBar">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="8" column="0" rowspan="2" colspan="2">
       <widget class="Line" name="line">
        <property name="lineWidth">
         <number>5</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QListWidget" name="baud_checkboxes">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>1000000</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">500000</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>400000</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>250000</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>200000</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>117647</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>57142</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>19230</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string>9615</string>
         </property>
         <property name="checkState">
          <enum>Checked</enum>
         </property>
        </item>
       </widget>
      </item>
      <item row="19" column="0" colspan="2">
       <widget class="QPushButton" name="abort_button">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Abort</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLabel" name="baud_label">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Baud rates to use:</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="minimum_label">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Minimum</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QLabel" name="label_5">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Connect to controller device:</string>
        </property>
       </widget>
      </item>
      <item row="13" column="1">
       <widget class="QSpinBox" name="maximum_spinBox">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximum">
         <number>253</number>
        </property>
        <property name="value">
         <number>96</number>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="maximum_label">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Maximum</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QComboBox" name="ports_list_combobox"/>
      </item>
      <item row="17" column="0" colspan="2">
       <widget class="QPushButton" name="scan_button">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>48</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>ArrowCursor</cursorShape>
        </property>
        <property name="toolTip">
         <string>Click here for awesomeness</string>
        </property>
        <property name="text">
         <string>Scan</string>
        </property>
        <property name="default">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="4" colspan="2">
     <widget class="QTreeWidget" name="motor_tree">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="frameShape">
       <enum>QFrame::HLine</enum>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="autoScroll">
       <bool>false</bool>
      </property>
      <property name="autoScrollMargin">
       <number>-25</number>
      </property>
      <property name="alternatingRowColors">
       <bool>false</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
      <property name="animated">
       <bool>false</bool>
      </property>
      <property name="expandsOnDoubleClick">
       <bool>false</bool>
      </property>
      <property name="columnCount">
       <number>3</number>
      </property>
      <column>
       <property name="text">
        <string notr="true">baudrate</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>id</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>model</string>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1158</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>returndelaytime_horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>returndelaytime_lcd</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>926</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>1074</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>torquemax_horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>torque_lcd</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>927</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>1075</x>
     <y>379</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maximum_dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>maximum_lcd</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1075</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>1075</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minimum_dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>minimum_lcd</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>927</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>927</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>position_dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>position_lcd</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>983</x>
     <y>525</y>
    </hint>
    <hint type="destinationlabel">
     <x>983</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>goal_dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>goal_lcd</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1075</x>
     <y>635</y>
    </hint>
    <hint type="destinationlabel">
     <x>1075</x>
     <y>723</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <designerdata>
  <property name="gridDeltaX">
   <number>10</number>
  </property>
  <property name="gridDeltaY">
   <number>10</number>
  </property>
  <property name="gridSnapX">
   <bool>true</bool>
  </property>
  <property name="gridSnapY">
   <bool>true</bool>
  </property>
  <property name="gridVisible">
   <bool>true</bool>
  </property>
 </designerdata>
</ui>
